### PR TITLE
Release - 1.0

### PR DESCRIPTION
**Pull request by nivida**
## Description

This is the stable release of v1.0.

### Compare View

[v1.0.0-beta.37 -> v1.0.0](https://github.com/ethereum/web3.js/compare/v1.0.0-beta.37...7e90479)

## Type of change

- [X] Release

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code. 
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no warnings.
- [x] I have updated or added types for all modules I've changed
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran ```npm run test``` in the root folder with success and extended the tests to cover all changes.
- [x] I ran ```npm run build``` in the root folder and tested it in the browser and with node.
- [x] I have tested my code with an ethereum test network.

### CODE REVIEW

File: packages/web3-providers-ws/package.json [hunk 1]
```diff
@@ -9,6 +9,6 @@
     "dependencies": {
         "underscore": "1.8.3",
         "web3-core-helpers": "1.0.0-beta.37",
-        "websocket": "git://github.com/frozeman/WebSocket-Node.git#browserifyCompatible"
+        "websocket": "1.0.28"
```
Comment (spalladino) [inline packages/web3-providers-ws/package.json]: The only addition of the `browserifyCompatible` branch in frozeman's fork seems to be [this change](https://github.com/frozeman/WebSocket-Node/commit/7004c39c42ac98875ab61126e5b4a925430f592c). I'm not sure why it was made in the first place, but are we sure that it is no longer needed?

File: packages/web3-core-promievent/package.json [hunk 2]
```diff
@@ -8,6 +8,6 @@
     "main": "src/index.js",
     "dependencies": {
         "any-promise": "1.3.0",
-        "eventemitter3": "1.1.1"
+        "eventemitter3": "3.1.2"
```
Comment (spalladino) [inline packages/web3-core-promievent/package.json:13]: I'm a bit worried about bumping a major version on a dependency, as it could introduce potential incompatibilities in something as critical as the promievent. Why was this needed?

**Review by spalladino — COMMENTED** [review]
Thank you so much for tackling this @nivida!! I'm super happy to see this happening.

I left two questions on dependencies changes. I'm worried about eventemitter3 in particular as we are shifting from 1.x to 3.x, but since I now little about that library, if you have already reviewed the changelog to make sure it's safe to update, I'm fine with it. 

As for websockets, I'm glad we are moving away from the github version since it was incorrectly packed and was giving a lot of headaches to many developers when installing it.

**Comment by alcuadrado** [comment]
I'm surprised by the number of dependencies upgrades to new major versions of this PR, especially because none of their uses has changed.

Does this version of the library have enough tests to do these upgrades with enough confidence that nothing breaks, @nivida? I'm not familiar enough with this codebase yet.

I agree with @spalladino that the `eventemitter3` is a special case, as it's exposed through promievents.

**Comment by sethfork** [comment]
Great to see the progress here @nivida! I'm curious what the 2.x branch/release is, as that seems to be a recent development.
> As for websockets, I'm glad we are moving away from the github version since it was incorrectly packed and was giving a lot of headaches to many developers when installing it.

+1 @spalladino. Can't wait to remove my preinstall scripts removing websocket git folders.

File: packages/web3-providers-ws/package.json (hunk 1 above)
Comment (nivida) [reply packages/web3-providers-ws/package.json]: I'm using the ``websocket`` package in the new architecture directly but will definitely do some additional tests. (There was an issue in the past with getting the global object with Firefox)

**Comment by nivida** [comment]
> Does this version of the library have enough tests to do these upgrades with enough confidence that nothing breaks, @nivida? I'm not familiar enough with this codebase yet.

Yes, I've downgraded the packages if the tests were failing. 

> I agree with @spalladino that the eventemitter3 is a special case, as it's exposed through promievents. 

Upgrading of the EventEmitter package doesn't have an impact on the functionality of Web3.js but I will definitely do some additional tests here too.

@sethfork I will publish an announcement which will explain anything :-)


Edit: I've checked the release notes of the EventEmitter package and the only breaking change they had was on the release of [2.0.0](https://github.com/primus/eventemitter3/releases/tag/2.0.0) which doesn't have an impact on Web3.js. @spalladino @alcuadrado

**Comment by wbt** [comment]
Is there any particular reason this version was picked, including the [security audit failure due to a tar dependency](https://github.com/trufflesuite/truffle/issues/1912) fixed in -beta38?  Is it because beta37 is the [last version to have had the pre-built Javascript files](https://github.com/ethereum/web3.js/issues/2623)?

**Comment by wbt** [comment]
Cross-linking to [comments by @sshelton76 on April 22](https://github.com/ethereum/web3.js/issues/2684#issuecomment-485537042)++ for convenient reference.

File: package.json [hunk 3]
```diff
@@ -2,7 +2,7 @@
   "name": "web3",
   "private": true,
   "namespace": "ethereum",
```
Comment (levino) [inline package.json]: I cannot find the key "namespace" in the specs...

**Comment by levino** [comment]
So. When release?

File: packages/web3-providers-ws/package.json [hunk 4]
```diff
@@ -8,6 +8,6 @@
     "dependencies": {
         "underscore": "1.9.1",
         "web3-core-helpers": "1.0.0-beta.37",
-        "websocket": "1.0.28"
+        "websocket": "github:frozeman/WebSocket-Node#browserifyCompatible"
```
Comment (Gudahtt) [inline packages/web3-providers-ws/package.json]: It looks like you've switched this back to the version that was incorrectly packed. That version includes a `.git` directory in the published module, which results in npm errors in certain cases (e.g. when running `npm install` a second time).

A few people in this PR already expressed relief at this error going away; it was the part I was most looking forward to about this release personally. Could you elaborate on why this update was reverted? I'd be happy to help out if you ran into an issue you don't have time to investigate.

File: packages/web3-providers-ws/package.json [hunk 5]
```diff
@@ -1,14 +1,13 @@
...
-    "repository": "https://github.com/ethereum/web3.js/tree/master/packages/web3-providers-ws",
+    "repository": "https://github.com/ethereum/web3.js/tree/1.x/packages/web3-providers-ws",
     "license": "LGPL-3.0",
     "main": "src/index.js",
     "dependencies": {
-        "underscore": "1.8.3",
+        "underscore": "1.9.1",
         "web3-core-helpers": "1.0.0-beta.37",
-        "websocket": "git://github.com/frozeman/WebSocket-Node.git#browserifyCompatible"
+        "websocket": "github:frozeman/WebSocket-Node#browserifyCompatible"
```
Comment (Gudahtt) [inline packages/web3-providers-ws/package.json:17]: It looks like you've switched this back to the version that was incorrectly packed. That version includes a `.git` directory in the published module, which results in npm errors in certain cases (e.g. when running `npm install` a second time).

A few people in this PR already expressed relief at this error going away; it was the part I was most looking forward to about this release personally. Could you elaborate on why this update was reverted? I'd be happy to help out if you ran into an issue you don't have time to investigate.

Comment (nivida) [reply packages/web3-providers-ws/package.json:17]: I've changed it back to the GitHub repository of @frozeman because I couldn't figure out why this change was required then. Was there a clear reason for it? @frozeman

Comment (Gudahtt) [reply packages/web3-providers-ws/package.json:17]: I found some additional context about that: https://github.com/theturtle32/WebSocket-Node/pull/314
I wonder if the issue this was meant to fix (https://github.com/theturtle32/WebSocket-Node/issues/304) is still an issue or not - that library has changed a great deal since that report. So has Edge, for that matter.

Comment (nivida) [reply packages/web3-providers-ws/package.json:17]: I've added the ``preinstall`` command to the ``package.json`` file of the ``web3-providers-ws`` module which will fix it. I will get in contact with the former maintainer to get the missing information.

Comment (Gudahtt) [reply packages/web3-providers-ws/package.json:17]: Great, thanks! That avoids the error I was seeing before at least.

I did test this on Edge (using browserify, as the PR described) just to see if it's still an issue, but I wasn't able to reproduce the problem. So I'm guessing whatever behavior that was meant to workaround in Edge 11 is no longer present in Edge 18.

Comment (nivida) [reply packages/web3-providers-ws/package.json:17]: The ``websocket`` package will get ignored for the minified file which will be used in the browser it just has an impact if it's running on node. Fabian just sent me the following explanation for it: 
``` 
There was a problem with the meteor build cloentnor snother packager. So I had to fork an makebthis change
It is important 
```

Comment (Gudahtt) [reply packages/web3-providers-ws/package.json:17]: The change in question was made solely to the `browser.js` file, so I don't believe it's a node issue per se - it was a problem with the browserify step itself, which of course is run using node. You can see [the commit](https://github.com/frozeman/WebSocket-Node/commit/7004c39c42ac98875ab61126e5b4a925430f592c) has the message `made browserify compat`.

So I believe the way I tried to reproduce it was correct. Though from the sounds of things, maybe it was something particular to that build process that resulted in the error, rather than just the browser used with the resulting bundle.

**Comment by alcuadrado** [comment]
> Edit: I've checked the release notes of the EventEmitter package and the only breaking change they had was on the release of [2.0.0](https://github.com/primus/eventemitter3/releases/tag/2.0.0) which doesn't have an impact on Web3.js. @spalladino @alcuadrado

It's a reasonable change. Thanks for looking into it.

File: packages/web3-eth-ens/package.json [hunk 6]
```diff
@@ -2,12 +2,12 @@
     "name": "web3-eth-ens",
     "version": "1.0.0-beta.37",
```
Comment (alcuadrado) [inline packages/web3-eth-ens/package.json]: Is this correct? Shouldn't all the packages be released as `1.0.0`?

**Review by alcuadrado — COMMENTED** [review]
I added a minor comment about the version of the internal packages. Sorry I haven't noticed this in my previous review.

File: packages/web3-eth-ens/package.json (hunk 6 above)
Comment (levino) [reply packages/web3-eth-ens/package.json]: Versions should not be bumped by hand. One should use `npx lerna publish` oder `npx lerna version`.

File: .travis.yml [hunk 7]
```diff
@@ -3,7 +3,6 @@ node_js:
     - "8"
     - "10"
     - "11"
-    - "12"
```
Comment (levino) [inline .travis.yml:6]: I think this should be documented / a follow up issue created. The background is, as far as I understand that `scrypt.js` does not support node version 12. We should also put this information in the `package.json` under the [`engine`](https://docs.npmjs.com/files/package.json#engines) field.

Also @nivida  please acknowledge that `scrypt.js` is [deprecated](https://github.com/axic/scrypt.js/issues/5#issuecomment-494846059) and needs to be removed asap. There will be no fix for it ever again!

Comment (nivida) [reply .travis.yml:6]: > I think this should be documented / a follow up issue created.

We already have an issue related to nodejs v12 https://github.com/ethereum/web3.js/issues/2913. 

> We should also put this information in the package.json under the engine field.

Thanks for the hint! I'll add the engine property.

> Also @nivida please acknowledge that scrypt.js is deprecated and needs to be removed asap. There will be no fix for it ever again!

I'm in contact with ethereum-js and I try to reach out to @axic to give us access to the scrypt.js repository and npm package. @alcuadrado opened already two PR's which will fix it in version 0.2.x and 0.3.x of the scrypt.js package. If axic doesn't want to give us the required access rights will we move the code of the ``scrypt.js`` package to Web3.js or we create a new NPM package which is carrying about the compatibility of it for older and newer nodejs versions.

**Comment by michaelsbradleyjr** [comment]
Hopefully PR #2938 can be considered prior to release.

**Comment by levino** [comment]
So why no release yet? What is the blocker?

**Comment by nivida** [comment]
@Levino Truffle has to run their tests to approve the stable release as it is. (@gnidan)

**Comment by levino** [comment]
Make them open a PR to run these tests here in CI.

**Review by frozeman — APPROVED** [review]

**Comment by frozeman** [comment]
Happy to see this. I’m especially happy that my last beta.37 was chosen for the 1.0, as it was my main goal to get this released, as the new API was my main effort to make it more intuitive. 
The 2.0 from @nividia can now contain breaking changes without that anybody gets surprised. :)

**Comment by michaelsbradleyjr** [comment]
I'm concerned I've found a bug on this PR branch (`release/1.0`). I have been working on #2938 and having some problems. I discovered by accident that I can't always decrypt wallets that I've encrypted with `web3-eth-accounts`. I switched from my PR branch back to `release/1.0` and found I had the same problem.

In the following workflow I'm using macOS 10.14.5, node 10.16.0, npm 6.10.1:

```
$ cd ~/repos/web3.js
$ git checkout origin/release/1.0
$ npm i
$ cd packages/web3-eth-accounts
$ npm pack
$ mkdir -p ~/temp/foo
$ cd ~/temp/foo
$ npm init -y
$ npm i ~/repos/web3.js/packages/web3-eth-accounts/web3-eth-accounts-1.0.0-beta.37.tgz
```
in a Node.js REPL:
```
$ node
> Accounts = require('web3-eth-accounts')
> ethAccounts = new Accounts()
> n = 256
> salt = '3a1012583f8be138537bc7cf8a50c925b6fcc01a9f7744c85a18fbdc07999f10'
> iv = Buffer.from('653195c3e2791ac53f3f19b125c18f8c', 'hex')
> uuid = Buffer.from('ff31ddc3e2791ac53f3f19b125c18fff', 'hex')
> pw = 'test'
> acc = ethAccounts.create()
> wallet1 = acc.encrypt(pw)
> ethAccounts.decrypt(wallet, pw)
{ address: '0x72387B35b2B84767Ab1b9756e8e640C468353CdF',
  privateKey:
   '0xcc53df7eeefdc84d172c59f69533a59d10dc36f1d14ffc7c576200beae8f3fe7',
  signTransaction: [Function: signTransaction],
  sign: [Function: sign],
  encrypt: [Function: encrypt] }
> wallet2 = acc.encrypt(pw, {n: n,salt: salt, iv: iv, uuid: uuid})
> ethAccounts.decrypt(wallet, pw)
Thrown:
Error: Key derivation failed - possibly wrong password
    at Accounts.decrypt (/Users/michael/temp/foo/node_modules/web3-eth-accounts/src/index.js:327:15)
```

The above is adapted from web3 1.0's [test suite](https://github.com/ethereum/web3.js/blob/release/1.0/test/eth.accounts.encrypt-decrypt.js#L128). It seems that if custom scrypt parameters are used during encryption then decryption may not succeed.

**Comment by gnidan** [comment]
@michaelsbradleyjr is the bug you've found introduced by this PR, or does it exist on 1.0.0-beta.37?

**Comment by michaelsbradleyjr** [comment]
@gnidan it seems to exist on this branch *and* with `npm i web3-eth-accounts@1.0.0-beta.37` from the registry.

I raised the matter here because there is a release planned for tomorrow, wanted to make sure to raise awareness prior to release.

**Comment by gnidan** [comment]
Ah, ok, in that case @michaelsbradleyjr: I vote that we don't fix additional bugs (bugs not introduced by this PR), since the primary motivator for this release is to enable further patching. I think instead we should just try to get PRs ready for 1.0.1 and the 2.0 alpha.

**Review by gnidan — APPROVED** [review]
PR https://github.com/trufflesuite/truffle/pull/2182 reasonably demonstrates that this release passes Truffle's CI. Approving this since it meets acceptance criteria :)

Woot! Thanks @nivida!

**Comment by iurimatias** [comment]
it's a 👍

1.0 here we come! Thanks @nivida

**Comment by nivida** [comment]
FYI

Bildschirmfoto 2019-07-13 um 13 12 18

**Comment by michaelsbradleyjr** [comment]
There's now a fix in #2938 for the [decrypt bug](https://github.com/ethereum/web3.js/pull/2905#issuecomment-511050537) — see [here](https://github.com/ethereum/web3.js/pull/2938/files#diff-7bb2a20126193b9ecfe4723f83429c49R354) and [here](https://github.com/ethereum/web3.js/pull/2938/files#diff-7bb2a20126193b9ecfe4723f83429c49R360). It still may be desirable to have that PR land in a future `1.x` patch, but I wanted to give a heads up.

**Comment by levino** [comment]
Nice. But not relevant for this PR. Please release, @nvidia. Just release it as `1.0.1` if need be.

**Comment by nivida** [comment]
@levino I‘m in contact with npm. We can‘t release 1.0 as 1.0.1 because 1.0.9 and 1.1.0 got published as well. I will ping them today again to get this process one step further. If npm isn’t answering or will reject the request will I release it as 1.2.0.

**Comment by levino** [comment]
There is no problem in releasing it as 1.1.1 or 1.2.0. Please stop delaying this. You can release it immediately. Just do it.

**Comment by michaelsbradleyjr** [comment]
If npm may make it possible for us to have `1.0.0` then I'd say it's worth waiting one more day. If by the end of today it's not possible to publish as `1.0.0` then `1.2.0` it is.

**Comment by levino** [comment]
Why? Because you like "nice numbers"? It does not matter that the version string esthetically pleases your eyes. It matters that it is logically correct and people can f***ing use a reliable and functioning piece of software. Will you also oppose the release of version 1.0.13 because the thirteen is bad luck?

**Comment by levino** [comment]
This "let's wait for npm" shit show has been going on for nine days already.

**Comment by levino** [comment]
Also there are **reasons** [why this is so hard](https://docs.npmjs.com/cli/unpublish#warning). It is against all best practices and **there is absolutely no need** to unpublish at all. There is not one single rational reason to not just publish it as `1.2.0`.

**Comment by Gudahtt** [comment]
While I don't think @Levino 's disrespectful tone is helpful, I do agree that unpublishing seems like a very bad idea. Any projects using the existing 1.0 or 1.1 release would be in for an unpleasant surprise.

Releasing this as 1.2.0 does seem to have a lot of benefits and no downsides.

**Comment by levino** [comment]
@Gudahtt Maybe it is not helpful, you are right. But then nobody here seems to be interested in making things work, so it cant harm either.

**Comment by wbt** [comment]
>Why? Because you like "nice numbers"?  

@Levino This project has been obsessed with a "1.0.0" release for a long while now.  As discussed [here](https://github.com/trufflesuite/truffle/issues/2140), this decision

>[seems](https://github.com/trufflesuite/truffle/issues/2140#issuecomment-507798120)   like a somewhat arbitrary marketing-driven versioning decision and probably a bit of a lie on the "stable release" language which seems more likely meant to satisfy some VC or manager or PR purpose than to necessarily have the appropriate technical backing. (To me, changing to 1.0.0 could just be used to signify a breaking change; it doesn't have to be conflated with completion of a major milestone or even stability in the release, as seems to be conflated in Web3. Hyperledger Fabric put their [first LTS release](https://hyperledger-fabric.readthedocs.io/en/release-1.4/whatsnew.html) at whatever version number they were at when they thought they were ready, specifically 1.4).

Neither @Gudahtt nor I [can see those downsides](https://github.com/ethereum/web3.js/pull/2905#issuecomment-513815856) to choosing a higher version number because of the lack of transparency, but there are pretty strong signals that such downsides exist at least in @nivida's view.  I don't think it's just aesthetics (and I don't think there's really a need for profanity either).  @nivida has consistently refused to share exactly what the specific reasons behind the decisions (like the super strong emphasis on 1.0.0) are, but they are clearly strong ones whatever they may be.  We will probably never really know, although disclosure would be much more consistent with open-source and the decentralization ethos this package is theoretically supposed to support.  I suspect the actual reason is just so much less consistent with that ethos that it would seem embarrassing to share the real reasons, and that's why there's such a strong desire to not explain the strong motivators behind going **specifically** to a "1.0.0 'stable' release."  

The [page](https://docs.npmjs.com/cli/unpublish#warning) you (@Levino) linked to warns that `unpublish` "is generally considered bad behavior."  [Abandoning semantic versioning](https://github.com/trufflesuite/truffle/issues/2140#issuecomment-507282380), putting out major breaking changes with number updates more granular than patch, keeping decision-making behind closed doors without even explaining reasons for conclusions, and rejecting [good-faith contributions](https://github.com/ethereum/web3.js/issues/2623#issuecomment-507273436) on the grounds that they weren't written by core contributors who prefer their own hypothetical versions, are too.  That apparently hasn't made a bit of difference; I doubt the linked-to warning will either.  I do not think the assertion that "nobody here seems to be interested in making things work" is accurate though, or even close.

I'm not convinced this project is ready for a "stable release" label/claim.  I *am* in favor of steps back toward semantic versioning and having some base version that can be patched/modified to resolve outstanding issues alongside appropriate bumps in the semver number depending on what gets changed, as things get changed, without burying @nivida or anybody else under a mountain of outstanding action items that *all* have to get changed at once before *any* new useful developments can be released.   In the absence of being able to actually take those steps, could we please at least find out more about the real reasons why not?

**Comment by nivida** [comment]
@wbt 
> Neither @Gudahtt nor I can see those downsides to choosing a higher version number because of the lack of transparency, but there are pretty strong signals that such downsides exist at least in @nivida's view. I don't think it's just aesthetics (and I don't think there's really a need for profanity either). @nivida has consistently refused to share exactly what the specific reasons behind the decisions (like the super strong emphasis on 1.0.0) are, but they are clearly strong ones whatever they may be. We will probably never really know, although disclosure would be much more consistent with open-source and the decentralization ethos this package is theoretically supposed to support. I suspect the actual reason is just so much less consistent with that ethos that it would seem embarrassing to share the real reasons, and that's why there's such a strong desire to not explain the strong motivators behind going specifically to a "1.0.0 'stable' release."

We discussed the situation and other topics in the [Ethereum Javascript Community (EJC) discord](https://discord.gg/NR42fy). The 1.2.0 release was seen as a worst-case solution. The worst-case solution wasn't taken because npm was answering on my support ticket and didn't reject the request as initially thought and "discussed" in the EJC. I've pinged them today and didn't get an answer until now and will do the release of 1.0 as 1.2.0 tomorrow (CEST). 

> The page you (@Levino) linked to warns that unpublish "is generally considered bad behavior." Abandoning semantic versioning, putting out major breaking changes with number updates more granular than patch, keeping decision-making behind closed doors without even explaining reasons for conclusions, and rejecting good-faith contributions on the grounds that they weren't written by core contributors who prefer their own hypothetical versions, are too. That apparently hasn't made a bit of difference; I doubt the linked-to warning will either. 

I'm maintaining this project currently for one year and the involvement of the community in decisions was always asked. I've asked for feedback in the personal blog post and the release announcement of the new architecture I've published and pull requests are welcomed and getting merged if the changes do have the quality I'm expecting. I've started to write down the first draft of a final 2.0 architecture [here](https://github.com/ethereum/web3-examples/pull/19) and hope many of you will be involved.

> I'm not convinced this project is ready for a "stable release" label/claim. I am in favor of steps back toward semantic versioning and having some base version that can be patched/modified to resolve outstanding issues alongside appropriate bumps in the semver number depending on what gets changed, as things get changed, without burying @nivida or anybody else under a mountain of outstanding action items that all have to get changed at once before any new useful developments can be released. 

The missing announcement will explain this closer and any further or required questions can be asked in the publicly accessible [EJC discord](https://discord.gg/NR42fy) or with an issue here on GitHub.

**Comment by wbt** [comment]
> We discussed the situation and other topics in the [Ethereum Javascript Community (EJC) discord](https://discord.gg/NR42fy). The 1.2.0 release was seen as a worst-case solution. 

Thanks for posting the Discord link.  I've just read through the complete history of the #web3js channel there. I found your messages saying "I would have to release 1.2 beacuse 1.1 is also existing" and "... [sic] and releasing of a 1.2.0 as initial stable release isn't something I will do" as well as "fuck off... I will release it as 1.2.0 because npm will probably anyways reject the request."  There is no other characterization of this as a "worst-case solution" nor any reasons given why it's bad.  It's just something you "won't do" for reasons that you consistently refuse to explain.  **WHY** do you think it's a "worst-case" solution?  What are the hidden downsides that we aren't seeing? 

> I'm maintaining this project currently for one year and the involvement of the community in decisions was always asked. I've asked for feedback in the personal blog post [no link] and the release announcement [no link] of the new architecture I've published and pull requests are welcomed and getting merged if the changes do have the quality I'm expecting. I've started to write down the first draft of a final 2.0 architecture here [no link] and hope many of you will be involved.

I maintain this [request](https://github.com/trufflesuite/truffle/issues/2140#issuecomment-507798120):
>that any discussions in a more rapid synchronous channel (like an in-person meeting, Discord, or teleconference) should be summarized on the archival listserv (now, it would be GitHub Issues), including who was there, what decisions were made, the reasons for them, and reasons for an alternative path that were considered but overridden. I would like to see more of that happening in these projects.

*...please!*

> > I'm not convinced this project is ready for a "stable release" label/claim....
> 
> The missing announcement will explain this closer and any further or required questions can be asked in the publicly accessible [EJC discord](https://discord.gg/NR42fy) or with an issue here on GitHub.

[Here](https://github.com/ethereum/web3.js/issues/2961) is that Issue.

**Comment by nivida** [comment]
@wbt Truffle and OpenZeppelin was the main actor to release beta.37 as 1.0 stable version of web3.js because it's used widely over the community. We were discussing this for a long time and I was first against this decision and would have liked to follow the path with the new architecture I've written since I've started here but this isn't possible because some projects were relying on the internal API and architecture of web3.js. Splitting this up between a 1.0 and 2.0 version of web3.js will give us the possibility to not break already existing DApp's. The new architecture in 2.0 does provide the same public API as we had before just with new core modules and some additional features but the real benefit of 2.0 is that we are able to use the latest features of JavaScript. Version 1.0 and also 2.0 will be maintained and improved by me and the community which wants to be involved.

I've locked the conversation of this PR because version 1.2.0 gets released as the initial non-beta version of the 1.0 architecture of web3.js tomorrow. Further discussion can be held over GitHub or the mentioned discord channel above.
